amplify/backend/function/eventNotifications/src/index.js

```diff
@@ -0,0 +1,198 @@
+
+/* Amplify Params - DO NOT EDIT
+	ENV
+	REGION
+	API_FITNESSPROJECTAPI_GRAPHQLAPIIDOUTPUT
+	API_FITNESSPROJECTAPI_GRAPHQLAPIENDPOINTOUTPUT
+Amplify Params - DO NOT EDIT */
+
+
+/**
+ * @type {import('@types/aws-lambda').APIGatewayProxyHandler}
+ */
+/** 
+exports.handler = event => {
+  console.log(`EVENT: ${JSON.stringify(event)}`);
+  for (const record of event.Records) {
+    console.log(record.eventID);
+    console.log(record.eventName);
+    console.log('DynamoDB Record: %j', record.dynamodb);
+  }
+  return Promise.resolve('Successfully processed DynamoDB record');
+};
+*/
+
+require("isomorphic-fetch");
+const gql = require("graphql-tag");
+
+const {
+  deleteEvent,
+} = require("/opt/mutations");
+
+const { getUser, eventsByDate, listEvents, getEvent, eventOwner  } = require("/opt/queries");
+
+const { client, sendNotification, s3 } = require("/opt/backendResources");
+const { loadCapitals } = require("/opt/stringConversion");
+const { SHA256 } = require("/opt/hash");
+const { WellArchitected } = require('aws-sdk');
+
+exports.handler = async (event, context, callback) => {
+  return await Promise.all (
+    event.Records.map(async (record) => {
+      console.log(event.Records);
+      if(record.eventName == "REMOVE"){
+        try {
+          const eventOwner= record.dynamodb.OldImage.userId.S;
+          console.log("event owner is " + eventOwner)
+          console.log("users in event are" + JSON.stringify(record.dynamodb.OldImage.users, null, 4))
+          // get name of event owner for push message 
+          const OwnerName = await client.query({
+            query: gql(getUser),
+            variables: {
+              id: eventOwner,
+            },
+          });
+
+          // hopfully gets list of all users in the post (OldImage == post before)
+          if (record.dynamodb.OldImage.users.L != null) {
+            console.log("inside if")
+            console.log(record.dynamodb.OldImage.users.L.length)
+            for (let i = 0; i < record.dynamodb.OldImage.users.L.length; i ++){
```

Comment:
We could use a .forEach() on the array instead. Also we should make a function for this since it's reused in multiple places